Source/CesiumRuntime/Private/CesiumGeoreference.cpp

```diff
@@ -57,83 +57,132 @@ createCoordinateSystem(const FVector& center, double scale) {
-    // Spawn georeference in the persistent level
-    FActorSpawnParameters spawnParameters;
-    spawnParameters.SpawnCollisionHandlingOverride =
-        ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
-    spawnParameters.OverrideLevel = world->PersistentLevel;
-    pGeoreference = world->SpawnActor<ACesiumGeoreference>(spawnParameters);
-    // Null check so the editor doesn't crash when it makes arbitrary calls to
-    // this function without a valid world context object.
-    if (pGeoreference) {
-      pGeoreference->Tags.Add(DEFAULT_GEOREFERENCE_TAG);
-    }
 
-  } else {
-    UE_LOG(
-        LogCesium,
-        Verbose,
-        TEXT("Using existing Georeference %s for actor %s"),
-        *pGeoreference->GetName(),
-        *WorldContextObject->GetName());
+  return Georeference;
+}
+
+ACesiumGeoreference*
+FindGeoreferenceWithDefaultName(const UObject* WorldContextObject) {
+  UWorld* World = WorldContextObject->GetWorld();
+  if (!IsValid(World))
+    return nullptr;
+
+  ACesiumGeoreference* Candidate = FindObject<ACesiumGeoreference>(
+      World->PersistentLevel,
+      TEXT("CesiumGeoreferenceDefault"));
+
+  // Test if PendingKill
+  if (IsValid(Candidate)) {
+    return Candidate;
   }
-  return pGeoreference;
+
+  return nullptr;
+}
+
+ACesiumGeoreference*
+CreateDefaultGeoreference(const UObject* WorldContextObject, const FName& Tag) {
+  UWorld* World = WorldContextObject->GetWorld();
+  if (!IsValid(World))
+    return nullptr;
+
+  // Spawn georeference in the persistent level
+  FActorSpawnParameters spawnParameters;
+  spawnParameters.SpawnCollisionHandlingOverride =
+      ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+  spawnParameters.OverrideLevel = World->PersistentLevel;
+
+  ACesiumGeoreference* Georeference =
+      World->SpawnActor<ACesiumGeoreference>(spawnParameters);
+  Georeference->Tags.Add(Tag);
+
+  return Georeference;
+}
+
+} // namespace
+
+/*static*/ const double ACesiumGeoreference::kMinimumScale = 1.0e-6;
+
+/*static*/ ACesiumGeoreference*
+ACesiumGeoreference::GetDefaultGeoreference(const UObject* WorldContextObject) {
+  if (!IsValid(WorldContextObject))
+    return nullptr;
+
+  ACesiumGeoreference* Georeference =
+      FindGeoreferenceWithTag(WorldContextObject, DEFAULT_GEOREFERENCE_TAG);
+  if (IsValid(Georeference))
+    return Georeference;
+
+  Georeference = FindGeoreferenceWithDefaultName(WorldContextObject);
+  if (IsValid(Georeference))
+    return Georeference;
+
+  Georeference =
+      CreateDefaultGeoreference(WorldContextObject, DEFAULT_GEOREFERENCE_TAG);
+
+  return Georeference;
+}
+
+/*static*/ ACesiumGeoreference*
+ACesiumGeoreference::GetDefaultGeoreferenceForActor(AActor* Actor) {
+  if (!IsValid(Actor))
+    return nullptr;
+
+  ACesiumGeoreference* Georeference = FindGeoreferenceAncestor(Actor);
+  if (IsValid(Georeference))
+    return Georeference;
+
+  Georeference = FindGeoreferenceWithTag(Actor, DEFAULT_GEOREFERENCE_TAG);
```

Comment:
Lines 175 - 185 look like the same logic in `::GetDefaultGeoreference`. Any change to just call that function instead? Would reduce duplication, but could also make it clearer what's happening... find an ancestor, if didn't find it, find it in the world